### ISSUE TITLE
Moral vagueness

### ISSUE THREAD

**Issue by modulesio** [Widdershin/programmers-oath#25]
> I will only undertake honest and moral work. I will stand firm against any requirement that exploits or harms people.

Almost everyone believes they have good morals, and the right amount of honesty for the limits imposed on them. The biggest exploiters spin their work as benefitting the receiver.

Is suspect almost everyone would agree they adhere to this regardless of what they do. Therefore this seems a null point.

Could this be more objective?

**Comment by bortzmeyer** [comment]
We can refer to texts like the Universal Declaration of Human Rights https://en.wikipedia.org/wiki/Universal_Declaration_of_Human_Rights They are not perfect (they are human work, not religious texts) but this is the closest we have to an objective set of morals. And it is signed by almost all nations in the world so are truly universal.

**Comment by mo-g** [comment]
I think it's fairly unlikely that there will ever be a 'Hague' to enforce the Oath, no matter how popular it gets. I feel like this serves as a command to signatories to stand up against evils when they are asked to commit them - to quit when your boss at cute-fluffy-web-connected-plushies-inc walks up with a man in black glasses and tells you all about your new project to put in a back door so the nice man from the FSB (CIA, GCHQ, etc) can listen in on what's being said around your kids toy.

TL;DR When you know it's wrong, but you do it anyway because you don't want to get fired.

But I'm not opposed to applying an objective standard - after all, I'm in favour of 'Do No Harm'. I'd be tempted to mark this as #needsstrenuousdebate or somesuch. :smile:

**Comment by bortzmeyer** [comment]
@mo-g The problem is that "you know it's wrong" depends too many on *personal* morals. For instance, a religious zealot may refuse to work on a software having something to do with abortion, while a feminist will consider his or her duty to work on the same software. Programmers should be careful not to abuse their (relative) power. [Quote here the mandatory Spiderman's uncle's saying about great powers and great responsabilities.] In the same way that a doctor must tend to someone sick, even if the doctor disapproves of this person's lifestyles or actions, a programmer should be careful of not letting his own personal opinions turn him into a non-elected non-responsible guy-in-charge. That's why *external* guidelines like the UHDR is useful.

**Comment by mo-g** [comment]
RIP Uncle Ben. :latin_cross:

So the argument really comes down to: Is personal morality a valid choice for a personal oath? I can see the risk if @Widdershin's goal of Google mandating this for new staff came true - anyone can argue their way out of any dispute by citing their personal morality. I would argue that both are needed. Universal secular morality (such as the UDHR, which isn't a bad starting point - though we should be wary of ongoing international disputes around that - As an example, LGBT+ coders might well find Article 16 contentious) gives an interpersonal standard we can hold each other to.

Enacting personal morality is also admittedly tricky - If someone were to refuse to sell a wedding cake to a gay couple because they hold conservative religious views, for example - under UK law, they can either choose to stop making wedding cakes - or to make it against their personal morality. Similarly, if my employer asked me to implement DRM in something I worked on, or to create GIS software for military use - refusing would realistically mean the end of my employment, and possibly a bad reference that could end my career as a programmer. 

This is a non-trivial issue for sure, and I still support the need for a comprehensive discussion before a change is made.

**Comment by bortzmeyer** [comment]
@mo-g About the UDHR: this is a pragmatic choice. True, the text is not perfect but there is *zero* chance to develop a better one today. (You can write a better text, but its recognition will be far behind the UDHR.) Remember it took a world-wide war to develop the UDHR. A better text will come only from a third world war or a world revolution. 

And article 16 just says that men and women has the right to marry, not that it has to be *one* man and *one* woman :-)

**Comment by mo-g** [comment]
Just highlighting potential things that ought to be thought about, that's all. ;)

**Comment by vassudanagunta** [comment]
@modulesio 

> The biggest exploiters spin their work as benefitting the receiver.

Spin by definition is dishonest. No amount of specifics in an Oath is going to curb people who are dishonest with others and themselves. 

As much as I'd like to add my specific morality into this Oath (I'd say that serving ads is immoral), it's better to simply get people to prioritize being moral and honest over making money. That is already a *huge* step forward in this ever selfist and materialistic world. 
 
It will be healthy to have *honest*  about what is moral. We're not even close to there, as is evident in the politics-as-war as opposed to politics-as-truth that rules the world today, and as evident in almost any discussion of how women are treated by men in technology (look at any post on the topic in Hacker News).
